### PR TITLE
docs: add info about dot files and dir in flat config

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
added the information that flat config doesn't ignore files or directories with name starts from dot.
some issue had been opened that mentions this #18214 and #18236

### CODE REVIEW

File: docs/src/use/configure/configuration-files.md [hunk 1]
```diff
@@ -173,6 +173,8 @@ This configuration specifies that all of the files in the `.config` directory sh
 
 For more information on configuring rules, see [Ignore Files](ignore).
 
+**Note:** Unlike legecy config, by default flat config doesn't ignore files or directories having a name starts with a dot, such as `.fileName` or `.directoryName`.
```
Comment (nzakas) [inline docs/src/use/configure/configuration-files.md:176]: A bit of cleanup:

```suggestion
::: important
Glob patterns always match files and directories that begin with a dot, such as `.foo.js` or `.fixtures`, unless those files are explicitly ignored. The only dot directory ignored by default is `.git`.
:::
```

Comment (Tanujkanti4441) [reply docs/src/use/configure/configuration-files.md]: Done!

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for @nzakas to verify.

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**